py_proxy/views.py

```diff
@@ -64,10 +72,29 @@ def content_type(request):
     return exc.HTTPFound(f"{via_url}/{url}")
 
 
-def _drop_from_url_begining(drop_chars, url):
-    """Drop drop_chars from begining of url."""
-    drop_before = len(drop_chars)
-    return url[drop_before:]
+def _generate_url_without_client_query_params(base_url, query_params):
+    """Generate a url without client query params.
+
+    :param base_url: the protocol, domain and path aka: https://thirdparty.url/foo.pdf
+    :type base_url: string
+
+    :param query_params: the query params to be added to the base_url
+    :type query_params: dict
+
+    :return: the url without the client query params
+    :rtype: string
+    """
+    client_params = [OPEN_SIDEBAR, CONFIG_FROM_FRAME]
+    filtered_params = "&".join(
+        [
+            f"{param}={value}"
+            for param, value in query_params.items()
+            if param not in client_params
+        ]
+    )
```

Comment:
Can we use urllib here? It might be a little safer when it comes to encoding and things.

```diff
diff --git a/py_proxy/views.py b/py_proxy/views.py
index a9e2f53..5ec5384 100644
--- a/py_proxy/views.py
+++ b/py_proxy/views.py
@@ -1,4 +1,6 @@
 """The views for the Pyramid app."""
+import urllib
+
 import pyramid.httpexceptions as exc
 import requests
 from pyramid import response, view
@@ -85,12 +87,9 @@ def _generate_url_without_client_query_params(base_url, query_params):
     :rtype: string
     """
     client_params = [OPEN_SIDEBAR, CONFIG_FROM_FRAME]
-    filtered_params = "&".join(
-        [
-            f"{param}={value}"
-            for param, value in query_params.items()
-            if param not in client_params
-        ]
+
+    filtered_params = urllib.parse.urlencode(
+        {key: value for key, value in query_params.items() if key not in client_params}
     )
     if filtered_params:
         return f"{base_url}?{filtered_params}"
```

I think this is what Via does too: <https://github.com/hypothesis/via/blob/986c2f772b0c5b1d8d7a4fdafb5a87362a834707/via/config_extractor.py#L48-L74>. Via actually has to parse the query params first (using `urlparse.parse_qsl(orig_qs, keep_blank_values=True)`) but py-proxy doesn't need to do that because it can just use Pyramid's `request.params`. But then, after popping the `via.*` query params, it looks like Via re-constructs the query string without them by [calling `urllib.urlencode()`](https://github.com/hypothesis/via/blob/986c2f772b0c5b1d8d7a4fdafb5a87362a834707/via/config_extractor.py#L67-L68).

Note that the organization of Python's standard `urllib` and `urlparse` libraries are different for Via (Python 2) and py-proxy (Python 3).